packages/client-sdk-web/src/utils/web-client-utils.ts

```diff
@@ -20,11 +20,24 @@ export function createCallMetadata(
 export function getWebControlEndpoint(
   credentialProvider: CredentialProvider
 ): string {
-  return `web.${credentialProvider.getControlEndpoint()}`;
+  if (credentialProvider.isControlEndpointOverridden()) {
+    return withProtocolPrefix(credentialProvider.getControlEndpoint());
+  }
+  return withProtocolPrefix(`${credentialProvider.getControlEndpoint()}`);
 }
 
 export function getWebCacheEndpoint(
   credentialProvider: CredentialProvider
 ): string {
-  return `web.${credentialProvider.getCacheEndpoint()}`;
+  if (credentialProvider.isCacheEndpointOverridden()) {
+    return withProtocolPrefix(credentialProvider.getCacheEndpoint());
+  }
+  return withProtocolPrefix(`${credentialProvider.getCacheEndpoint()}`);
+}
+
+function withProtocolPrefix(endpoint: string): string {
+  if (endpoint.match(/^http(?:s)?:\/\//)) {
```

Comment:
Should we make this a case-insensitive regex match?

edit: I see where you inline the protocol later